src/base/bittorrent/tracker.cpp

```diff
@@ -392,26 +392,27 @@ void Tracker::prepareAnnounceResponse(const TrackerAnnounceRequest &announceReq)
     };
 
     // peer list
-    // [BEP-7] IPv6 Tracker Extension (partial support)
+    // [BEP-7] IPv6 Tracker Extension (partial support - only the part that concerns BEP-23)
     // [BEP-23] Tracker Returns Compact Peer Lists
     if (announceReq.compact) {
-        lt::entry::list_type peerList;
-        lt::entry::list_type peer6List;
+        lt::entry::string_type peers;
+        lt::entry::string_type peers6;
 
         int counter = 0;
         for (const Peer &peer : asConst(torrentStats.peers)) {
             if (counter++ >= announceReq.numwant)
                 break;
 
-            if (peer.endpoint.size() == 6)  // IPv4
-                peerList.emplace_back(peer.endpoint);
-            else if (peer.endpoint.size() == 18)  // IPv6
-                peer6List.emplace_back(peer.endpoint);
+            if (peer.endpoint.size() == 6)  // IPv4 + port
```

Comment:
please leave the "port" out of the comment, the meaning is already embedded when using `peer.endpoint`.